src/gui/qgstemporalcontrollerdockwidget.cpp

```diff
@@ -0,0 +1,238 @@
+  connect( mBackButton, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::backButton_clicked );
+  connect( mNextButton, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::nextButton_clicked );
+  connect( mPreviousButton, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::previousButton_clicked );
+  connect( mStopButton, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::stopButton_clicked );
+
+  connect( mStartDateTime, &QDateTimeEdit::dateTimeChanged, this, &QgsTemporalControllerDockWidget::updateTemporalExtent );
+  connect( mEndDateTime, &QDateTimeEdit::dateTimeChanged, this, &QgsTemporalControllerDockWidget::updateTemporalExtent );
+  connect( mSpinBox, qgis::overload<int>::of( &QSpinBox::valueChanged ), this, &QgsTemporalControllerDockWidget::updateFrameDuration );
+  connect( mTimeStepsComboBox, qgis::overload<int>::of( &QComboBox::currentIndexChanged ), this, &QgsTemporalControllerDockWidget::updateFrameDuration );
+  connect( mSlider, &QSlider::valueChanged, this, &QgsTemporalControllerDockWidget::timeSlider_valueChanged );
+
+  connect( mNavigationObject, &QgsTemporalNavigationObject::updateTemporalRange, this, &QgsTemporalControllerDockWidget::updateSlider );
+
+  connect( mSettingsDialog->mapSettingsWidget(), &QgsTemporalMapSettingsWidget::frameRateChanged, this, &QgsTemporalControllerDockWidget::updateFrameRate );
+
+  connect( mSettings, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::settings_clicked );
+  connect( mSetToProjectTimeButton, &QPushButton::clicked, this, &QgsTemporalControllerDockWidget::setDatesToProjectTime );
+
+  init();
+}
+
+QgsTemporalControllerDockWidget::~QgsTemporalControllerDockWidget()
+{
+}
+
+void QgsTemporalControllerDockWidget::init()
+{
+  QgsDateTimeRange range;
+
+  if ( QgsProject::instance()->timeSettings() )
+    range = QgsProject::instance()->timeSettings()->temporalRange();
+  QLocale locale;
+
+  mStartDateTime->setDisplayFormat( locale.dateTimeFormat() );
+  mEndDateTime->setDisplayFormat( locale.dateTimeFormat() );
+
+  if ( range.begin().isValid() && range.end().isValid() )
+  {
+    mStartDateTime->setDateTime( range.begin() );
+    mEndDateTime->setDateTime( range.end() );
+  }
+
+  mSetToProjectTimeButton->setToolTip( tr( "Set to project time" ) );
+
+  QStringList listSteps = ( QStringList() << tr( "Seconds" ) << tr( "Minutes" ) << tr( "Hours" )
+                            << tr( "Days" ) <<
+                            tr( "Months" ) << ( "Years" ) );
+
+  mTimeStepsComboBox->addItems( listSteps );
+  mTimeStepsComboBox->setCurrentIndex( 2 );
+
+  mSpinBox->setMinimum( 1 );
+  mSpinBox->setSingleStep( 1 );
+  mSpinBox->setValue( 1 );
+  mSpinBox->setEnabled( true );
+
+  updateFrameRate();
+  updateTemporalExtent();
+  updateFrameDuration();
+}
+
+void QgsTemporalControllerDockWidget::updateFrameRate()
+{
+  if ( mSettingsDialog && mSettingsDialog->mapSettingsWidget() )
```

Comment:
Again, don't make the settings dialog a member -- we only need to create it when it's being shown, and then copy its settings back to mNavigationObject after that

Reply:
this function is called when there is a change of value in the input rame rate box from the dialog, how will we know which dialog to get the input if we don't store it?